### ISSUE TITLE
KingfisherOptions Bug

### ISSUE THREAD

**Issue by Limon-catch** [onevcat/Kingfisher#123]
It seems that there is a bug in Kingfisher.

I tried to set two KingfisherOptions, but it doesn't work.

I created a [OptionSetTypePlayground](https://github.com/Limon-catch/OptionSetTypePlayground) to test it.

```swift
let optionsInfos: KingfisherOptionsInfo = [.Options: [KingfisherOptions.CacheMemoryOnly, KingfisherOptions.ForceRefresh]]

illustrationImageView.kf_setImageWithURL(NSURL(), placeholderImage: nil, optionsInfo: optionsInfos)



```

But **Kingfisher** never uses `optionsInfos `, because `optionsInfos` might be an `Array`, Kingfisher doesn't take it into consideration.

```swift
func parseOptionsInfo(optionsInfo: KingfisherOptionsInfo?) -> String {

    // If optionsInfo?[.Options] is a Array? 
    // Just like [KingfisherOptions.CacheMemoryOnly, KingfisherOptions.ForceRefresh]

    if let optionsInOptionsInfo = optionsInfo?[.Options] as? KingfisherOptions {
        return "\(optionsInOptionsInfo)"

    } else {
        return "DefaultOptions"
    }
}
```

**Comment by onevcat** [comment]
:( 
It seems to be a side-effect for the Swift 2 type inference. I will give it a fix soon.

Thanks for pointing it out.

**Comment by onevcat** [comment]
@Limon-catch 
Before a fix on it, you could use `as KingfisherOptions` to get a temporary workaround:

```swift
let optionsInfos: KingfisherOptionsInfo = [.Options: [KingfisherOptions.CacheMemoryOnly, KingfisherOptions.ForceRefresh] as KingfisherOptions]

```

**Comment by Limon-catch** [comment]
👌

**Comment by onevcat** [comment]
Fixed in #127 

Please notice that the original usage of `KingfisherOptionsInfo` changed a little. It is now safer and better. It is fairly easy to fix it. Just change your original code:

```swift
let optionsInfos: KingfisherOptionsInfo = 
[
    .Options: [KingfisherOptions.CacheMemoryOnly, KingfisherOptions.ForceRefresh], 
    .Transition: ImageTransition.Fade(1)
]
```

to something like this:

```swift
let optionsInfos: KingfisherOptionsInfo = 
[
    .Options([KingfisherOptions.CacheMemoryOnly, KingfisherOptions.ForceRefresh]), 
    .Transition(ImageTransition.Fade(1))
]
```
